src/hooks/web3.ts

```diff
@@ -49,25 +62,39 @@ export function useEagerConnect() {
  * and out after checking what network theyre on
  */
 export function useInactiveListener(suppress = false) {
+  const dispatch = useAppDispatch()
   const { active, error, activate } = useWeb3ReactCore() // specifically using useWeb3React because of what this hook does
+  const referralLink = useReferralLink()
 
   useEffect(() => {
     const { ethereum } = window
 
     if (ethereum && ethereum.on && !active && !error && !suppress) {
       const handleChainChanged = () => {
         // eat errors
-        activate(injected, undefined, true).catch((error) => {
-          console.error('Failed to activate after chain changed', error)
-        })
+        activate(injected, undefined, true)
+          .catch((error) => {
+            console.error('Failed to activate after chain changed', error)
+          })
+          .finally(() => {
+            if (referralLink) {
+              dispatch(updateAffiliateLink({ affiliateLink: referralLink }))
+            }
+          })
       }
 
       const handleAccountsChanged = (accounts: string[]) => {
         if (accounts.length > 0) {
           // eat errors
-          activate(injected, undefined, true).catch((error) => {
-            console.error('Failed to activate after accounts changed', error)
-          })
+          activate(injected, undefined, true)
+            .catch((error) => {
+              console.error('Failed to activate after accounts changed', error)
+            })
+            .finally(() => {
+              if (referralLink) {
+                dispatch(updateAffiliateLink({ affiliateLink: referralLink }))
+              }
+            })
```

Comment:
I still don't understand why do we need this finally clauses. Can you elaborate on why this file needs to be updated?

Reply:
I did not use useEffect before so we required this code. But now I am using useEffect, so don't need to specify the code in this file. I've removed it.